WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -56,8 +57,9 @@ public async Task ConfirmConnectionAsync(TimeSpan connectionConfirmationTimeout,
 
 		private async Task<bool> TryConfirmConnectionAsync(CancellationToken cancellationToken)
 		{
-			var inputVsize = Constants.P2wpkhInputVirtualSize;
-			var inputRemainingVsizes = new[] { ProtocolConstants.MaxVsizePerAlice - inputVsize };
+			var inputVsize = Coin.ScriptPubKey.EstimateInputVsize();
+			var totalInputVsize = ArenaClient.VsizeCredentialClient.MaxAmount;
```

Comment:
Hmm, although this wasn't introduced here (the removed code makes the same equivalence), but using `MaxAmount` seems a bit strange to me here. For readability, if the property used here indicate that not only is this the max amount, it's also allocated and distributed for each Alice, so using `PerAliceVsizeAllocation` seems better, as it gives some explanation for why this starts at an arbitrary number instead of 0, which is a bit unintuitive.

As written this is still correct IMO, the only thing forcing these parameters to be equal prevents is if we want to differentially issue vsize credentials, which doesn't sound useful at all.

Reply:
hmm, @lontivero:

>  (Also the MaxAmountPerAlice just in case one day we need to change it for any reason like performance problems).

what issues did you have in mind? i could only think of ways in which letting these values diverge reduces performance.. technically the allocation can be up to `K * MaxAmount`, which i guess means we can save 1 bit per range proof. is that what you meant? if so, then `AliceClient` should get the allocation amount parameter explicitly, because `totalInputVsize` should start with that value, not the range proof constrained value.

Reply:
It is always better to use parameters instead of constants (when it is possible and when it makes sense) because that creates a less fragile protocol, otherwise any change on any of those values creates an incompatible protocol version that requires clients to upgrade. In this case the range proof width depends on the MaxAmout that today is quite big. In case one day we need to change that value we can do it without breaking everything. For example, imagine we realize that the MaxValue was chosen ridiculously  big and nobody has never register a 4000 BTC coin, instead the bigger coin registered was 100BTC, in this case we can reduce the range proof width many bits what saves bandwidth and coordinator resources. However that is only possible if the clients don't have the parameters harcoded.

Reply:
isn't this an argument for keeping the max value and the allocation value separate?